### PR TITLE
Filter out global concurrency limits used for deployment concurrency limiting

### LINKED ISSUE #15316: Deployment-backed global concurrency limits should not be mixed in with user-created GCL's
### Describe the current behavior

Deployment-based concurrency limits implicitly create and use global concurrency limits behind-the-scenes for enforcement. When a worker pulls down a scheduled flow run that's part of a deployment with a set `deployment.concurrency_limit`, a special global concurrency limit object will be created on behalf of the user for use in limiting the deployment's runs. These limits have a special pattern name `deployment:{deployment.id}` but are otherwise the same as if a user were to have created one via the `concurrency` helpers. 

This has the potential to create issues as these "system-owned" resources are exposed to users in the UI and via the prefect CLI which:
1. can create confusion and clutter for a user wanting to view GCL's that they create/manage
2. can create unexpected data issues as users can freely modify the underlying GCL's creating unexpected side effects for enforcing the deployment's desired concurrency limit

### Describe the proposed behavior

Because prefect creates these objects on behalf of users, we shouldn't expose it to users in the same way that user-defined global concurrency limits are. 

deployment-related global concurrency limits should be hidden:
- in the UI on the concurrency page in the global concurrency limits table
- from the CLI via prefect gcl ls
- from CLI actions like: prefect gcl enable/disable/delete/update/inspect
- bonus: guard users from creating a gcl with a name like deployment:<uuid>

Implementation note:
Rather than filter by name prefixes, we should add a new field to concurrency limits for better tagging and filtering of these deployment-related concurrency limit objects as system-owned.

In particular, consider adding an extendable `meta: JSONB` column or an enumerated column `used_for` to be used to filter out non-user-created GCL's from the GCL web apis.

### Example Use

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by collincchoy**
Closes #15316 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

**Review by zzstoatzz — COMMENTED** [review]
just a prelim comment

but why an ENUM if we only have one member as opposed to your other suggestion in the issue JSONB?

**Comment by collincchoy** [comment]
> just a prelim question
> 
> but why an ENUM if we only have one member as opposed to your other suggestion in the issue JSONB? naively it seems like JSONB would be more forgiving / not require extra migrations

I was leaning toward a JSON value initially for the flexibility but went this route just for the simpler querying/indexing story (imagining at least _some_ complexity from supporting sqlite+pg) and to get PR feedback sooner. 

Hmm yeah lmk what you think tho - I'd be happy to switch stuff over if that's what we want here.

ah and yeah the single-value enum seems odd I suppose but even with a JSON type, would still only have 1 literal value or None for now.

File: src/prefect/server/schemas/core.py [hunk 1]
```diff
@@ -654,6 +663,10 @@ class ConcurrencyLimitV2(ORMBaseModel):
     avg_slot_occupancy_seconds: float = Field(
         default=2.0, description="The average amount of time a slot is occupied."
     )
+    used_for: Optional[InternalConcurrencyLimitUsedFor] = Field(
```
Comment (cicdw) [inline src/prefect/server/schemas/core.py:666]: Hmmm having this field be something clients need to be responsible for feels off if it's purpose is to signify a "private" limit -- were other design alternatives considered for this that I could compare against?

**Comment by collincchoy** [comment]
See close notes on https://github.com/PrefectHQ/prefect/issues/15316

**Pull request closed without merging**